org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/processing/JobManager.java

```diff
@@ -558,4 +561,25 @@ public String toString() {
 		}
 		return buffer.toString();
 	}
+
+	protected synchronized boolean findLastJob(Predicate<IJob> testJob) {
```

Comment:
This change / code can be further simplified with the merging of same code block above.